### ISSUE TITLE
[Vim and Git] trustExitCode Git option

### ISSUE THREAD

**Issue by mzanibelli** [vim-advent-calendar/vimways.org#20]
`:cquit` can also be used to abort a diff or a merge when `difftool.trustExitCode` and `mergetool.trustExitCode` are set to `true`. I think it should be documented in the article.

**Comment by swalladge** [comment]
I documented `:cquit` in the article. I didn't mention `trustExitCode` partly because I _think_ it's the default behaviour and partly because I don't set those options myself. Can you confirm cases where `trustExitCode` needs to be explicitly set to `true` for exiting with a non-zero status to be classed as an abort action?

**Comment by chrisbra** [comment]
as mentioned on reddit, I am not sure about difftool, but at least for mergetool, this behaviour is the default since git 2.11.1 (see commit https://github.com/gitster/git/commit/2967284456e9053ee8)

**Comment by romainl** [comment]
2.11.1 was released almost two years ago. We should decide whether:

- we should assume everyone is using an up-to-date Git and keep the article as is,
- or we should take a defensive approach and mention that requirement in the text.

**Comment by mzanibelli** [comment]
After a few tests: it's *not* on by default when using `difftool` with git 2.19.2. The manpage is correct:
> git-difftool invokes a diff tool individually on each file. Errors reported by the diff tool are ignored by default.

**Comment by mzanibelli** [comment]
Interesting finding:

* `difftool.trustExitCode` must be set in the `[difftool]` section and is not enabled by default
* `mergetool.trustExitCode` must be set in the `[mergetool "mymergetool"]` section if using a custom mergetool. And it's *not* on by default: the commit states that it's enabled for the standard `vimdiff` tool only. The article uses a custom tool.

TL; DR: still relevant to document

**Comment by romainl** [comment]
OK, it looks like the article needs some editing.

**Comment by swalladge** [comment]
Yep you're correct. I can't understand why I haven't run across that before. Sorry for the confusion. I'll update the article asap.
